packages/engine/Source/Scene/Picking.js

```diff
@@ -373,24 +398,130 @@ Picking.prototype.pickVoxelCoordinate = function (
+ * @param {PickedMetadataInfo} pickedMetadataInfo Information about the picked metadata.
+ * @returns The metadata values
+ *
+ * @private
+ */
+Picking.prototype.pickMetadata = function (
+  scene,
+  windowPosition,
+  pickedMetadataInfo
+) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("windowPosition", windowPosition);
+  Check.typeOf.object("pickedMetadataInfo", pickedMetadataInfo);
+  //>>includeEnd('debug');
+
+  const { context, frameState, defaultView } = scene;
+  const { viewport, pickFramebuffer } = defaultView;
+
+  scene.view = defaultView;
+
+  viewport.x = 0;
+  viewport.y = 0;
+  viewport.width = context.drawingBufferWidth;
+  viewport.height = context.drawingBufferHeight;
+
+  let passState = defaultView.passState;
+  passState.viewport = BoundingRectangle.clone(viewport, passState.viewport);
+
+  const drawingBufferPosition = SceneTransforms.transformWindowToDrawingBuffer(
+    scene,
+    windowPosition,
+    scratchPosition
+  );
+  const drawingBufferRectangle = computePickingDrawingBufferRectangle(
+    context.drawingBufferHeight,
+    drawingBufferPosition,
+    1.0,
+    1.0,
+    scratchRectangle
+  );
+
+  scene.jobScheduler.disableThisFrame();
+
+  scene.updateFrameState();
+  frameState.cullingVolume = getPickCullingVolume(
+    scene,
+    drawingBufferPosition,
+    drawingBufferRectangle.width,
+    drawingBufferRectangle.height,
+    viewport
+  );
+  frameState.invertClassification = false;
+
+  frameState.passes.pick = true;
+  frameState.tilesetPassState = pickTilesetPassState;
+
+  // Insert the information about the picked metadata property
+  // into the frame state, so that the `Scene.updateDerivedCommands`
+  // call can detect any changes in the picked metadata description,
+  // and update the derived commands for the new picked metadata
+  // property
+  frameState.pickingMetadata = true;
+  frameState.pickedMetadataInfo = pickedMetadataInfo;
+  context.uniformState.update(frameState);
+
+  scene.updateEnvironment();
+
+  passState = pickFramebuffer.begin(drawingBufferRectangle, viewport);
+
+  scene.updateAndExecuteCommands(passState, scratchColorZero);
+
+  // When OIT is enabled, then the resolveFrameBuffers function
+  // will juggle around several frame buffers, and eventually use
+  // the "environmentState.originalFramebuffer" instead of the
+  // picking frame buffer. Skipping a million questions, just
+  // switch OIT off here:
+  const oldOIT = scene._environmentState.useOIT;
+  scene._environmentState.useOIT = false;
+  scene.resolveFramebuffers(passState);
+  scene._environmentState.useOIT = oldOIT;
+
+  const rawMetadataPixel = pickFramebuffer.readCenterPixel(
+    drawingBufferRectangle
+  );
+  context.endFrame();
+
+  frameState.pickingMetadata = false;
+
+  const metadataValue = MetadataPicking.decodeMetadataValues(
+    pickedMetadataInfo.classProperty,
+    rawMetadataPixel
+  );
+
+  return metadataValue;
+};
+
+/**
+ * @typedef {object} PickedMetadataInfo
```

Comment:
This just followed the pattern that was used for other types, like (random example) the `ContextOptions`. These appear in other classes, and are marked as `unresolved` in VSCode, but they still appear in the docs, e.g. at https://cesium.com/learn/cesiumjs/ref-doc/global.html#ContextOptions  

I also don't like some of the related glitches, quirks, and corner cases for the "typing" here. However, I'll move it into a "class" (still marked as `@private`), I hope that's OK.

Reply:
Apparently not. I can `import` that for VSCode. But then it will complain about the unused import. I'll leave it as it is. Sorting out general shortcomings of the JS-ecosystem is not in the scope of this PR.

Reply:
@jjhembd  You can open an issue for that, also referring to all the other `typedef`'s that already exist.

Reply:
The point below convinced me. I now moved it into an own "class". The question about the `unresolved` in VSCode still applies (for this and many other types).